lib/spack/spack/repo.py

```diff
@@ -1362,6 +1362,33 @@ def get_pkg_class(self, pkg_name):
         if not inspect.isclass(cls):
             tty.die("%s.%s is not a class" % (pkg_name, class_name))
 
+        new_cfg_settings = (
+            spack.config.get("packages").get(pkg_name, {}).get("package_attributes", {})
+        )
+        overidden_attrs = getattr(cls, "overidden_attrs", {})
```

Comment:
Can you add a comment before this line making clear that this is to clear any previous changes to ensure they don't go stale? Also I think for performance it would be good if we could avoid doing this if the config hasn't changed.

Reply:
> for performance it would be good if we could avoid doing this if the config hasn't changed.

How do we know whether the config has changed (I don't think we have a means to track that)?

Reply:
We could store a hash of the previous config, but that might be more bother than it's worth. We should test the performance with this PR and see if we need to do something about it